### DISCUSSION TITLE
How to get microsecond precision over NTP

### DISCUSSION THREAD

**Discussion by LordOfDeadbush** [earlephilhower/arduino-pico#2443] [Q&A]
Hello,
I was wondering if it is possible to get the time in microseconds with an NTP sync?

I've been looking at the docs and it seems that I can only get precision to the second. 

Been using this for reference:
[https://arduino-pico.readthedocs.io/en/latest/wifintp.html](url)

**Comment by dquadros** [comment]
NTP can, at best, give milliseconds precision. This involves determining and accounting for the round-trip times, and they vary a lot if you are connecting to a NTP server over the internet. Note also that most implementations are of SNTP, not NTP. SNTP use the same protocol but with a compatible but simpler  packet format.

References:
https://en.wikipedia.org/wiki/Network_Time_Protocol
https://datatracker.ietf.org/doc/html/rfc4330 (this is not the latest documentation, but I found it easier to read the RFC 5905

**Reply by LordOfDeadbush** [reply]
Hello,
Thanks for the reply! I am using my own NTP server on LAN (over ethernet only with the W5500 EVB Pico) so I think the lag should be somewhat negligible. 

Do you have any suggestions on what to do instead?

Corn
